app/src/main/java/org/jellyfin/androidtv/ui/playback/VideoManager.java

```diff
@@ -437,37 +432,43 @@ public void disableSubs() {
         }
     }
 
+    private int offsetStreamIndex(int index, boolean adjustByAdding, @Nullable List<MediaStream> allStreams) {
+        if (index < 0 || allStreams == null)
+            return -1;
+
+        for (MediaStream stream : allStreams) {
+            if (!stream.getIsExternal())
+                break;
+            index += adjustByAdding ? 1 : -1;
+        }
+
+        return index < 0 || index >= allStreams.size() ? -1 : index;
+    }
+
     public boolean setSubtitleTrack(int index, @Nullable List<MediaStream> allStreams) {
-        if (!nativeMode && allStreams != null) {
-            //find the relative order of our sub index within the sub tracks in VLC
-            int vlcIndex = 1; // start at 1 to account for "disabled"
-            for (MediaStream stream : allStreams) {
-                if (stream.getType() == MediaStreamType.Subtitle && !stream.getIsExternal()) {
-                    if (stream.getIndex() == index) {
-                        break;
-                    }
-                    vlcIndex++;
-                }
-            }
+        if (isNativeMode() || allStreams == null)
+            return false;
 
-            org.videolan.libvlc.MediaPlayer.TrackDescription vlcSub;
-            try {
-                vlcSub = getSubtitleTracks()[vlcIndex];
-
-            } catch (IndexOutOfBoundsException e) {
-                Timber.e("Could not locate subtitle with index %s in vlc track info", index);
-                return false;
-            } catch (NullPointerException e) {
-                Timber.e("No subtitle tracks found in player trying to set subtitle with index %s in vlc track info", index);
-                return false;
-            }
+        int vlcIndex = offsetStreamIndex(index, false, allStreams);
+
+        if (vlcIndex < 0)
+            return false;
 
-            Timber.i("Setting Vlc sub to %s", vlcSub.name);
-            return mVlcPlayer.setSpuTrack(vlcSub.id);
+        org.videolan.libvlc.MediaPlayer.TrackDescription vlcSub = null;
+        for (org.videolan.libvlc.MediaPlayer.TrackDescription subTrack: mVlcPlayer.getSpuTracks()) {
```

Comment:
Why use full package names instead of importing MediaPlayer here?

Reply:
That's a copy & paste from reordering the old code 😅. I'll change it to an import